deployment/document-understanding-solution.template

```diff
@@ -0,0 +1,701 @@
+            Action:
+              - "logs:CreateLogGroup"
+              - "logs:CreateLogStream"
+              - "logs:PutLogEvents"
+            Resource:
+              - !Join [
+                  "",
+                  [
+                    "arn:aws:logs:",
+                    Ref: "AWS::Region",
+                    ":",
+                    Ref: "AWS::AccountId",
+                    ":log-group:/aws/lambda/",
+                    !Ref CICDHelper,
+                    ":*",
+                  ],
+                ]
+          - Effect: "Allow"
+            Action:
+              - "codecommit:CreateCommit"
+              - "codecommit:PutFile"
+              - "codecommit:GetBranch"
+            Resource:
+              - !Sub ${DocumentUnderstandingRepo.Arn}
+          - Effect: "Allow"
+            Action:
+              - "apigateway:Delete*"
+            Resource:
+              - !Sub "arn:aws:apigateway:${AWS::Region}::*"
+          - Effect: "Allow"
+            Action:
+              - "cloudformation:DeleteStack"
+              - "cloudformation:Describe*"
+              - "cloudformation:Delete*"
+            Resource:
+              - !Sub "arn:aws:cloudformation:${AWS::Region}:${AWS::AccountId}:*"
+          - Effect: "Allow"
+            Action:
+              - "cognito-idp:AdminDeleteUser"
+              - "cognito-idp:DeleteUser"
+              - "cognito-idp:DeleteUserPool"
+              - "cognito-idp:DeleteUserPoolClient"
+            Resource:
+              - !Sub "arn:aws:cognito-idp:${AWS::Region}:${AWS::AccountId}:*"
+          - Effect: "Allow"
+            Action:
+              - "cognito-identity:DeleteIdentityPool"
+              - "cognito-identity:SetIdentityPoolRoles"
+            Resource:
+              - !Sub "arn:aws:cognito-identity:${AWS::Region}:${AWS::AccountId}:*"
+          - Effect: "Allow"
+            Action:
+              - "dynamodb:DeleteTable"
+              - "dynamodb:Describe*"
+            Resource:
+              - !Sub "arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:*"
+          - Effect: "Allow"
+            Action:
+              - "es:DeleteElasticsearchDomain"
+              - "es:DescribeElasticsearchDomain"
+            Resource:
+              - !Sub "arn:aws:es:${AWS::Region}:${AWS::AccountId}:domain/*"
+          - Effect: "Allow"
+            Action:
+              - "iam:Delete*"
+              - "iam:Describe*"
+              - "iam:DetachRolePolicy"
+            Resource:
+              - !Sub "arn:aws:iam::${AWS::AccountId}:*"
+          - Effect: "Allow"
+            Action:
+              - "lambda:GetEventSourceMapping"
+              - "lambda:DeleteFunction"
+              - "lambda:InvokeFunction"
+              - "lambda:RemovePermission"
+              - "lambda:DeleteLayerVersion"
+              - "lambda:DeleteEventSourceMapping"
+            Resource:
+              - !Sub "arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:*"
```

Comment:
Almost all our resources such as Lambda, DynamoDB, S3 ,SQS , SNS begin with the StackName prefix. Is there a way to import the stack name (I believe its currently V081*) and use it in this template.
I am wondering if we should go a step further and use the prefix when we specify the ARN structure.
For example
"arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function/{StackName}*

This is not a blocker for the deployment but we can avoid over permissive policies

Reply:
We definitely can import the stack name, there is already a variable defined, called StackName, and we can change it to import in the same way as the source bucket or version. I have opened a ticket to address reducing permissions, and I think we should tackle this in that ticket.